app/controllers/files_controller.rb

```diff
@@ -0,0 +1,20 @@
+class FilesController < ApplicationController
+  def upload
+    if File.file?(Upload.base_upload_dir.join(params[:path]))
+      case File.extname(params[:path]).downcase
+      when ".jpg", ".jpeg", ".png", ".gif"
+        disp = "inline"
+      else
+        disp = "attachment"
+      end
+      mime = mime_type(params[:path])
+      
+      send_file Upload.base_upload_dir.join(params[:path]).to_s,
```

Comment:
I checked that, actually: `<img src=...>` is fine, and the image will show up.  But there's one case where it's less fine: If a student submits a large image that doesn't fit on the screen, then the file-viewing UI will show it compressed to fit, with a magnifying-glass icon to pop it out.  That is simply a `<a target="_blank" href="image path">`, and that will open a new tab showing the image.  If content-disposition=attachment, though, it'll open a new tab and instantly close it and offer a Download File dialog, which is not the ideal experience.